### PR TITLE
fix: only consider persisted columns for simple operations

### PR DESCRIPTION
**Pull request by patricebender**
it doesnt make sense to `UPDATE`, `DELETE` or to `INSERT` a virtual element, which only exists during runtime, such as:

- `virtual` elements
- calculated elements
- associations

### CODE REVIEW

**Comment by BobdenOs** [comment]
Might be solved by: https://github.com/cap-js/cds-dbs/pull/408

**Comment by BobdenOs** [comment]
Looks like the update column check is not working correctly. As the snapshot tests show all update columns being removed. Probably the check should be inverted

**Review by BobdenOs — APPROVED** [review]

**Comment by BobdenOs** [comment]
@patricebender I guess now that the test is fixed the title should be updated?

**Comment by patricebender** [comment]
> @patricebender I guess now that the test is fixed the title should be updated?

If this is the proper fix, I'd be glad to merge this. The pr was originally only to showcase the issue.

File: db-service/lib/cqn2sql.js [hunk 1]
```diff
@@ -749,8 +749,9 @@ class CQN2SQLRenderer {
     if (_with) _add(_with, x => this.expr(x))
     function _add(data, sql4) {
       for (let c in data) {
-        if (!elements || (c in elements && !elements[c].virtual)) {
-          if (cds.unfold && elements?.[c].is_struct) continue // skip structs from universal csn
+        const columnExistsInDatabase =
+          elements && c in elements && !elements[c].virtual && !elements[c].isAssociation && !elements[c].value
```
[the remark this answers is not in the document]
Comment (patricebender) [reply db-service/lib/cqn2sql.js:753]: If `elements` is `undefined` we'd get a type error:
```js
> 'bar' in elements
VM240:1 Uncaught TypeError: Cannot use 'in' operator to search for 'bar' in undefined
    at <anonymous>:1:7
```

**Pull request merged**